src/main/osd/osd.c

```diff
@@ -806,10 +814,21 @@ static void osdRefreshStats(void)
     osdShowStats(osdStatsRowCount);
 }
 
-static void osdShowArmed(void)
+static timeDelta_t osdShowArmed(void)
 {
+    static bool everArmed = false;
```

Comment:
I wonder if this could be replaced by checking `ARMING_FLAG(WAS_EVER_ARMED)`. It will need some refactoring, but I can have a look into it.

Reply:
No, I originally tried that. But by the time we get to this code we are already armed and `ARMING_FLAG(WAS_EVER_ARMED)` is already true.

Reply:
See #9248.